fury/ui/core.py

```diff
@@ -730,23 +732,33 @@ def __init__(
             Adds text shadow.
         size : (int, int)
             Size (width, height) in pixels of the text bounding box.
+        auto_font_scale : bool, optional
+            Automatically scale font according to the text bounding box.
+        dynamic_bbox : bool, optional
+            Automatically reisize the bounding box according to the content.
         """
         super(TextBlock2D, self).__init__(position=position)
         self.scene = None
         self.have_bg = bool(bg_color)
-        if size is not None:
-            self.resize(size)
-        else:
-            self.font_size = font_size
         self.color = color
         self.background_color = bg_color
         self.font_family = font_family
-        self.justification = justification
+        self._justification = justification
         self.bold = bold
         self.italic = italic
         self.shadow = shadow
-        self.vertical_justification = vertical_justification
+        self._vertical_justification = vertical_justification
+        self.auto_font_scale = auto_font_scale
+        if self.auto_font_scale:
+            self.actor.SetTextScaleModeToProp()
+        self.dynamic_bbox = dynamic_bbox
```

Comment:
Is it not necessary to do a setter for these parameters, or are they supposed to be defined only when creating the instance of the object?